### PR TITLE
Make `ReproduciblePropertiesWriter` entry order independent

### PR DESCRIPTION
**Pull request by bamboo**
By writing it on top of `PropertiesUtils` in order to make the parameter name index files reproducible.

### CODE REVIEW

File: buildSrc/subprojects/build-base/src/main/kotlin/org/gradle/build/ReproduciblePropertiesWriter.kt [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                writer.write("# $comment")
+                writer.write(lineSeparator)
+            }
+            writer.write(content)
+        }
+    }
+
+    fun store(data: Map<String, *>) = store(propertiesFrom(data))
+
+    fun store(properties: Properties): String =
```
Comment (eskatos) [inline buildSrc/subprojects/build-base/src/main/kotlin/org/gradle/build/ReproduciblePropertiesWriter.kt]: :x: this function and the one above are named `store` but they don't store, they just transform properties to a string, this is confusing

File: buildSrc/subprojects/build-base/src/main/kotlin/org/gradle/build/ReproduciblePropertiesWriter.kt [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        toString(properties)
```
Comment (wolfs) [inline buildSrc/subprojects/build-base/src/main/kotlin/org/gradle/build/ReproduciblePropertiesWriter.kt]: Why don't you use `PropertiesUtils.store`?

Comment (eskatos) [reply buildSrc/subprojects/build-base/src/main/kotlin/org/gradle/build/ReproduciblePropertiesWriter.kt]: Very good point. I think we simply missed `PropertiesUtils`.

Comment (bamboo) [reply buildSrc/subprojects/build-base/src/main/kotlin/org/gradle/build/ReproduciblePropertiesWriter.kt]: Thanks for the hint, @wolfs.

File: buildSrc/subprojects/build/src/main/groovy/org/gradle/build/ReproduciblePropertiesWriter.groovy [hunk 3]
```diff
@@ -13,38 +13,27 @@
...
 package org.gradle.build
 
+import com.google.common.base.Charsets
+import org.gradle.internal.util.PropertiesUtils
+
 class ReproduciblePropertiesWriter {
 
-    /**
-     * Writes {@link Map} of data as {@link Properties} to a file, but without including the timestamp comment.
-     */
```
Comment (eskatos) [inline buildSrc/subprojects/build/src/main/groovy/org/gradle/build/ReproduciblePropertiesWriter.groovy]: :o: lost javadoc

Comment (bamboo) [reply buildSrc/subprojects/build/src/main/groovy/org/gradle/build/ReproduciblePropertiesWriter.groovy]: Fixed.

**Pull request merged**